yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -462,65 +582,100 @@ func (p *ParserIssueDetector) ParseAndProcessDDL(query string) error {
 			if len(alter.ConstraintColumns) > 0 {
 				p.addConstraintAsIndex(alter.SchemaName, alter.TableName, alter.ConstraintColumns, alter.ConstraintName)
 			}
+
+			// Track PK columns
+			tm := p.getOrCreateTableMetadata(alter.GetObjectName())
+
+			// Add to Constraints slice for comprehensive tracking
+			pkConstraint := ConstraintMetadata{
+				Type:    "PRIMARY",
+				Name:    alter.ConstraintName,
+				Columns: append([]string{}, alter.ConstraintColumns...),
+			}
+			tm.Constraints = append(tm.Constraints, pkConstraint)
 		}
 
 		if alter.ConstraintType == queryparser.UNIQUE_CONSTR_TYPE {
 			// Process unique constraint as index for foreign key detection
 			if len(alter.ConstraintColumns) > 0 {
 				p.addConstraintAsIndex(alter.SchemaName, alter.TableName, alter.ConstraintColumns, alter.ConstraintName)
 			}
+			// Track UNIQUE columns
+			if len(alter.ConstraintColumns) > 0 {
+				qualifiedTable := alter.GetObjectName()
+				tm := p.getOrCreateTableMetadata(qualifiedTable)
+				uniqueConstraint := ConstraintMetadata{
+					Type:    "UNIQUE",
+					Name:    alter.ConstraintName,
+					Columns: append([]string{}, alter.ConstraintColumns...),
+				}
+				tm.Constraints = append(tm.Constraints, uniqueConstraint)
+			}
+		}
+		// Track NOT NULL alter commands
+		if len(alter.SetNotNullColumns) > 0 {
+			qualifiedTable := alter.GetObjectName()
+			tm := p.getOrCreateTableMetadata(qualifiedTable)
+			for _, col := range alter.SetNotNullColumns {
+				if colMeta, exists := tm.Columns[col]; exists {
+					colMeta.IsNotNull = true
+				}
+			}
+		}
+		if len(alter.DropNotNullColumns) > 0 {
```

Comment:
this is just for safety, correct? ideally we wouldn't expect DROP statements in the pg_dump.